src/Components/Components/src/PersistentState/ComponentStatePersistenceManager.cs

```diff
@@ -177,6 +187,9 @@ private void InferRenderModes(Renderer renderer)
     }
 
     internal Task<bool> TryPauseAsync(IPersistentComponentStateStore store)
+        => TryPauseAsync(store, DefaultPersistenceReason.Instance);
+
+    internal Task<bool> TryPauseAsync(IPersistentComponentStateStore store, IPersistenceReason persistenceReason)
```

Comment:
Do not create and overload. Simply change this method and adjust all the places it's used